sklearn/neighbors/base.py

```diff
@@ -300,11 +303,27 @@ class from an array representing our data set and ask who's
         if self._fit_method is None:
             raise ValueError("must fit neighbors before querying")
 
-        X = check_array(X, accept_sparse='csr')
-
         if n_neighbors is None:
             n_neighbors = self.n_neighbors
 
+        if X is not None:
+            query_is_train = False
+            X = check_array(X, accept_sparse='csr')
+        else:
+            query_is_train = True
+            X = self._fit_X
+            # Include an extra neighbor in order to avoid marking the sample
```

Comment:
It's a confusing comment, whatever intended. "Include an extra neighbor to account for the sample itself being returned, which is removed later" or something might be clearer.